src/lib/utils/cookieUtils.ts

```diff
@@ -0,0 +1,40 @@
+import * as cookie from 'cookie'
+import { COOKIE_NAME } from '$lib/utils/constants'
+import { mapInputToObject } from '$lib/mapper/mapper'
+
+export function updateCookie(data: string, currentPageTitle: string, oldCookie?: string): string {
+  const cookies = cookie.parse(oldCookie || '')
+  const formData = cookies[COOKIE_NAME]
+  const pages = formData ? formData.split('|') : []
+  const currentCookieIndex = pages.findIndex((p) => p.includes(currentPageTitle))
+  const pageData = `${currentPageTitle}:${data}`
+  if (currentCookieIndex === -1) {
+    pages.push(pageData)
+  } else {
+    pages[currentCookieIndex] = pageData
+  }
+  return cookie.serialize(COOKIE_NAME, pages.join('|'), {
+    secure: true,
+    httpOnly: true,
+  })
+}
+
+export function removeCookie(cookieName: string): string {
+  return cookie.serialize(cookieName, {})
+}
+
+export function cookieToForm(oldCookie: string): Record<string, string> {
+  const cookies = cookie.parse(oldCookie)
+  const formCookies = cookies[COOKIE_NAME]
+  if (formCookies) {
+    const pageCookies = formCookies.split('|')
+    return pageCookies.reduce((acc, p) => {
+      const [key, value] = p.split(':')
+      return {
+        ...acc,
+        [key]: mapInputToObject(value),
+      }
+    }, {})
+  }
+  return {}
```

Comment:
I think it looks good! But here is a suggestion of a slightly more FP way to do it. Is it simpler or not?

```suggestion
  return (cookies[COOKIE_NAME] ?? '')
    .split('|')
    .filter((cookie) => cookie.length > 0)
    .reduce((acc, p) => {
      const [key, value] = p.split(':')
      
      return {
        ...acc,
        [key]: mapInputToObject(value),
      }
    }, {})
```